lib/tokenable.rb

```diff
@@ -0,0 +1,12 @@
+module Tokenable
+  extend ActiveSupport::Concern
+
+  included do
+    before_create :generate_token
```

Comment:
There is one problem with this solution, and thats generating the token in before_create. Its a huge security threat, because whoever would steal your database, could access all accounts knowing the token. Change the pw and have access to them. For that reason, you should create token only when user requests the reset and preferably expire it within an hour or 2.